src/app/newSettings/Sections/General/components/Appearance.tsx

```diff
@@ -55,13 +55,25 @@ const Appearance = () => {
 
   // CHECK IF THIS WORKS BEFORE MERGE
   useEffect(() => {
+    function isAppearanceExist(prevAppearances, theme, img) {
+      return prevAppearances.some((appearance) => appearance.theme === theme && appearance.img === img);
+    }
+
+    function addAppearanceIfUnique(prevAppearances, theme, img) {
+      if (isAppearanceExist(prevAppearances, theme, img)) {
+        return prevAppearances;
+      }
+      return [...prevAppearances, { theme, img }];
+    }
+
+    const updateAppearance = (theme, img) => {
+      setAppearances((prevAppearances) => addAppearanceIfUnique(prevAppearances, theme, img));
+    };
+
     isStarWarsThemeAvailable(plan)
       .then((isStarWarsThemeAvailable) => {
-        if (
-          !appearances.some((appearance) => appearance.theme === 'starwars' && appearance.img === appearance_dark) &&
-          isStarWarsThemeAvailable
-        ) {
-          setAppearances([...appearances, { theme: 'starwars', img: appearance_dark }]);
+        if (isStarWarsThemeAvailable) {
+          updateAppearance('starwars', appearance_dark);
```

Comment:
This is not maintainable, if you keep adding Themes the file can become incredibly large